cas-server-core/src/test/java/org/jasig/cas/util/http/HttpMessageTest.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a
+ * copy of the License at the following location:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.util.http;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import org.junit.Test;
+
+/**
+ *
+ * @author Francesco Cina
+ * @since 4.1
+ */
+public class HttpMessageTest {
+
+	@Test
+	public void verifyAsyncArgIsTakenIntoAccount() throws Exception {
+		assertTrue(new HttpMessage(null, "").isAsynchronous());
+		assertTrue(new HttpMessage(null, "", true).isAsynchronous());
```

Comment:
Also, mock up urls and sample test. This is weird. Your test sort of violates API contract because these parameters should never be null or blank. You could use a test url.

In fact, please a separate test for null checks and blank checks too!